### PR TITLE
Harden CI

### PR DESCRIPTION
**Pull request by Krzmbrzl**
A couple of CI additions that will hopefully make CI detect things like undefined behavior, memory leaks, etc. Hence, things like #323 should be caught sooner.

### CODE REVIEW

**Comment by Krzmbrzl** [comment]
@evaleev The long-running Wick tests all fail once I enable them (and get rid of the non-compiling parts) by evaluating to constants that are at least two orders of magnitude larger than the expected ones.
I'm not sure whether or not this is an actual problem or not. Maybe you could have a look?

Besides, I would tend to move such extremely long running tests into the benchmarks (stripping them from their test functionality and instead only use for timing purposes)…

https://github.com/ValeevGroup/SeQuant/blob/db8b3f04905e063d1bbbe49fccf0f5519d716cb8/tests/unit/test_wick.cpp#L750-L808

File: tests/unit/test_mbpt_cc.cpp [hunk 1]
```diff
@@ -33,85 +32,87 @@ TEST_CASE("mbpt_cc", "[mbpt/cc]") {
...
-      REQUIRE(t_eqs.size() == N + 1);
-      for (auto k = 0; k <= N; ++k) {
-        REQUIRE(t_eqs[k]);
-      }
-
-      // std::wcout << to_latex_align(t_eqs[0], 5, 3) << std::endl;
-      if (C == 3 && ansatz == CC::Ansatz::U) {
-        REQUIRE(size(t_eqs[0]) == 56);
-      }
+  SECTION("eom_cc"){SECTION("EOM-CCSD"){const auto N = 2;
```
Comment (ajay-mk) [inline tests/unit/test_mbpt_cc.cpp:35]: Formatting issue?

File: tests/integration/CMakeLists.txt [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
+            # Single-Reference closed-shell Coupled-Cluster equation generation (fast spin-traced)
+            "stcc.cpp"
+            # Single-Reference closed-shell Coupled-Cluster equation generation (rigorous spin-traced)
+            "stcc_rigorous.cpp"
+        )
+    endif()
+else()
+    list(APPEND TEST_CASES
+        # Single-Reference Coupled-Cluster equation generation (reduced test set)
+        "srcc.cpp ->  |2 t csv sf"
```
[the remark this answers is not in the document]
Comment (Krzmbrzl) [reply tests/integration/CMakeLists.txt:26]: I don't think that's super important. Given the fact that you have to explicitly disable long tests, most folks will just run all tests, which includes the stcc ones.

File: tests/unit/test_mbpt_cc.cpp (hunk 1 above)
Comment (Krzmbrzl) [reply tests/unit/test_mbpt_cc.cpp:35]: I completely agree. I guess that this is some sort of bug/edge case in clang-format :eyes:

File: tests/integration/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ajay-mk) [reply tests/integration/CMakeLists.txt:26]: Yeah, I thought `stcc` could catch potential issues in the spin-tracing path under sanitizers, but the unit tests cover it already. It's probably fine, not a big deal.

**Comment by Krzmbrzl** [comment]
When researching this I definitely found that compiler optimizations bear the potential of confusing valgrind, producing invalid outputs. I haven't found anything specifically about `-Og` but given every source effectively says "disable optimizations`, I would err on the safe side and keep them disabled entirely.
